plotly_resampler/aggregation/aggregation_interface.py

```diff
@@ -4,169 +4,260 @@
-            List containing the regex matching the supported datatypes, by default None.
+        x_dtype_regex_list: List[str], optional
+            List containing the regex matching the supported datatypes for the x array,
+            by default None.
+        y_dtype_regex_list: List[str], optional
+            List containing the regex matching the supported datatypes for the y array,
+            by default None.
 
         """
         self.interleave_gaps = interleave_gaps
-        self.dtype_regex_list = dtype_regex_list
-        self.nan_position = nan_position.lower()
-        super().__init__()
+        self.x_dtype_regex_list = x_dtype_regex_list
+        self.y_dtype_regex_list = y_dtype_regex_list
 
-    @abstractmethod
-    def _aggregate(self, s: pd.Series, n_out: int) -> pd.Series:
-        pass
+    @staticmethod
+    def _calc_med_diff(x_agg: np.ndarray) -> Tuple[float, np.ndarray]:
+        # ----- divide and conquer heuristic to calculate the median diff ------
+        # remark: thanks to the prepend -> x_diff.shape === len(s)
+        x_diff = np.diff(x_agg, prepend=x_agg[0])
+
+        # To do so - use an approach where we reshape the data
+        # into `n_blocks` blocks and calculate the mean and then the median on that
+        # Why use `median` instead of a global mean?
+        #   => when you have large gaps, they will be represented by a large diff
+        #      which will skew the mean way more than the median!
+        n_blocks = 128
+        if x_agg.shape[0] > 5 * n_blocks:
+            blck_size = x_diff.shape[0] // n_blocks
+
+            # convert the index series index diff into a reshaped view (i.e., sid_v)
+            sid_v: np.ndarray = x_diff[: blck_size * n_blocks].reshape(n_blocks, -1)
+
+            # calculate the min and max and calculate the median on that
+            med_diff = np.median(np.mean(sid_v, axis=1))
+        else:
+            med_diff = np.median(x_diff)
+
+        return med_diff, x_diff
+
+    @staticmethod
+    def _get_gap_mask(x_agg: np.ndarray) -> Optional[np.ndarray]:
+        # ------- INSERT None between gaps / irregularly sampled data -------
+        med_diff, s_idx_diff = AbstractAggregator._calc_med_diff(x_agg)
+
+        # TODO: this 4 was revealed to me in a dream, but it seems to work well
+        gap_mask = s_idx_diff > 4 * med_diff
+        if not any(gap_mask):
+            return
+        return gap_mask
+
+    @staticmethod
+    def insert_gap_none(
+        x_agg: np.ndarray,
+        y_agg: np.ndarray,
+        idxs: np.ndarray,
+    ) -> Tuple[np.ndarray, np.ndarray]:
+        """Insert None values in the y_agg array when there are gaps.
+
+        Parameters
+        ----------
+        x_agg: np.ndarray
+            The x array. This is used to determine the gaps.
+        y_agg: np.ndarray
+            The y array. A copy of this array will be expanded with None values where
+            there are gaps.
+        idxs: np.ndarray
```

Comment:
Types are already specified in the signature - no need to duplicate in the docs.